src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -86,6 +87,20 @@ internal static void ExtractSegmentIdentifierAndParenthesisExpression(string seg
                 identifier = segmentText;
                 parenthesisExpression = null;
             }
+            else if (segmentText[segmentText.Length - 1] == ':')
+            {

```

Comment:
add a comment describing logic -- is this to support syntax such as:
 entityset(key):/escape-function-path?